go/libraries/doltcore/schema/index.go

```diff
@@ -54,6 +54,10 @@ type Index interface {
 	// ToTableTuple returns a tuple that may be used to retrieve the original row from the indexed table when given
 	// a full index key (and not a partial index key).
 	ToTableTuple(ctx context.Context, fullKey types.Tuple, format *types.NomsBinFormat) (types.Tuple, error)
+	// GetPrefixLengths returns the prefix lengths for the index
+	GetPrefixLengths() []uint16
+	// SetPrefixLengths applies to prefix lengths
+	SetPrefixLengths(prefixLengths []uint16)
```

Comment:
I did think of this, but I saw that the index creation code had a lot of uses and wanted to avoid changing a bunch of functions 😅.

Regardless, should `PrefixLengths` be in another argument or should it go in `IndexProperties`?
For example, should 
```go
func NewIndex(name string, tags, allTags []uint64, indexColl IndexCollection, props IndexProperties) Index
```
change to 
```go
func NewIndex(name string, tags, allTags []uint64, prefixLengths []uint16, indexColl IndexCollection, props IndexProperties) Index
```
or should `IndexProperties` just have another field?

I'm in favor of adding it as a new parameter.

Reply:
We should just change all of those functions lol. I’m in favor of a new parameter rather than another field, but if there are too many parameters, we could move everything to index properties.